### PR TITLE
Docs: Update postgres protocol SSL & auth section

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: blackbox/docs/protocols/postgres.txt [hunk 1]
```diff
@@ -38,15 +38,34 @@ CrateDB emulates PostgreSQL server version ``9.5``
 Start-Up
 --------
 
-The connection handshake is implemented without authentication support. CrateDB
-will always respond with ``AuthenticationOK`` and any SSLRequests are answered
-with ``N`` to indicate that it does not support SSL.
+SSL Support
+...........
+
+SSL support is only available in the `Enterprise Edition`_. If enterprise is
```
Comment (mikethebeer) [inline blackbox/docs/protocols/postgres.txt]: probably link to `SSL/TLS Configuration for the PostgreSQL Wire Protocol` page.

File: blackbox/docs/protocols/postgres.txt [hunk 2]
```diff
@@ -38,15 +38,34 @@ CrateDB emulates PostgreSQL server version ``9.5``
...
+disabled, any ``SSLRequests`` are always answered with ``N``, indicating to a
+client that SSL is not available.
+
+
+Authentication
+..............
+
+If the `Enterprise Edition`_ is enabled, authentication methods can be
```
Comment (mikethebeer) [inline blackbox/docs/protocols/postgres.txt]: same here: link to `Host-Based Authentication` page

File: blackbox/docs/protocols/postgres.txt [hunk 3]
```diff
@@ -38,15 +38,34 @@ CrateDB emulates PostgreSQL server version ``9.5``
...
 --------
 
-The connection handshake is implemented without authentication support. CrateDB
-will always respond with ``AuthenticationOK`` and any SSLRequests are answered
-with ``N`` to indicate that it does not support SSL.
+SSL Support
+...........
+
+SSL support is only available in the `Enterprise Edition`_. If enterprise is
+disabled, any ``SSLRequests`` are always answered with ``N``, indicating to a
```
Comment (m-kharbat) [inline blackbox/docs/protocols/postgres.txt]: this phrase doesn't sound correct, i checked with @Autophagy. as a suggestion you can use : `If the Enterprise Edition is disabled, all SLLRequests are answered with N, indicating to the client that SSL is not available.`

File: blackbox/docs/protocols/postgres.txt [hunk 4]
```diff
@@ -38,15 +38,34 @@ CrateDB emulates PostgreSQL server version ``9.5``
...
+
+
+Authentication
+..............
+
+If the `Enterprise Edition`_ is enabled, authentication methods can be
+configured using :doc:`/administration/hba`.
+
+If the enterprise functionality is not enabled there won't be any
+authentication and any "start up" requests will be answered with
```
Comment (m-kharbat) [inline blackbox/docs/protocols/postgres.txt]: same here

**Comment by mfussenegger** [comment]
Closing, I split this into https://github.com/crate/crate/pull/5796 and https://github.com/crate/crate/pull/5801

**Pull request closed without merging**